opba-banking-protocol-facade/src/test/java/de/adorsys/opba/protocol/facade/services/authorization/ServiceSessionAndAuthSessionTest.java

```diff
@@ -0,0 +1,244 @@
+                                                  .requestId(listAccountsResponse.getXRequestId())
+                                                  .build()
+                       )
+                       .scaAuthenticationData(authData)
+                       .build();
+    }
+
+    private ListAccountsRequest buildListAccountRequest(UUID sessionId) {
+        return ListAccountsRequest.builder()
+                       .facadeServiceable(
+                               FacadeServiceableRequest.builder()
+                                       .bankId(TEST_BANK_ID)
+                                       .requestId(REQUEST_ID)
+                                       .serviceSessionId(sessionId)
+                                       .sessionPassword(PASSWORD)
+                                       .fintechRedirectUrlOk(REDIRECT_URL_OK)
+                                       .fintechRedirectUrlNok(REDIRECT_URL_NO_OK)
+                                       .build()
+                       ).build();
+    }
+
+    private void assertAllSessions(UUID sessionId) {
+        Optional<AuthSession> authenticationSessionOptional = authenticationSessions.findById(sessionId);
+        Optional<ServiceSession> serviceSessionOptional = serviceSessionRepository.findById(sessionId);
+
+        assertThat(authenticationSessionOptional.isPresent()).isTrue();
+        assertThat(serviceSessionOptional.isPresent()).isTrue();
+
+        AuthSession authenticationSession = authenticationSessionOptional.get();
+        ServiceSession serviceSessionFromAuth = authenticationSession.getParent();
+        ServiceSession serviceSessionFromDB = serviceSessionOptional.get();
+
+        assertThat(serviceSessionFromDB.getId()).isEqualTo(serviceSessionFromAuth.getId());
+        assertThat(serviceSessionFromDB.getAuthSession().getId()).isEqualTo(authenticationSession.getId());
+        assertThat(serviceSessionFromDB.getAuthSession().getRedirectCode()).isEqualTo(authenticationSession.getRedirectCode());
+    }
+
+    private AuthorizationRequiredResult buildAuthorizationRequiredResult() {
+        return new AuthorizationRequiredResult(URI.create("http://localhost:4400/account-information/login?encryptedConsentId=ZwaFosuf-GopQSJwEcWKIn708zVNbrHvmWZh6amuM59aBEMABLWcMZATy6v9CghY5hpqT3HlUQAcRciH8hB3asz9MpaJIQIH3NJX8IHgetw=_=_psGLvQpt9Q&redirectId=7154aa62-b744-4072-b4b9-0fa87048e2c8"));
+    }
+
+    private ValidationErrorResult buildValidationErrorResultResult() {
+        return new ValidationErrorResult(URI.create("http://localhost:5500/parameters/provide-more/761b551d-5980-11ea-a703-acde48001122?q=%5B%7B%22uiCode%22:%22boolean.boolean%22,%20%22ctxCode%22:%22consent.recurringIndicator%22,%20%22message%22:%22%7Bno.ctx.recurringIndicator%7D%22%7D,%20%7B%22uiCode%22:%22date.string%22,%20%22ctxCode%22:%22consent.validUntil%22,%20%22message%22:%22%7Bfuture.ctx.validUntil%7D%22%7D%5D"));
```

Comment:
Makes no sense to embed this large URL 
`http://localhost:5500/parameters/provide-more/`
is completely enough